mtags/src/main/scala-2/scala/meta/internal/pc/SemanticTokenProvider.scala

```diff
@@ -36,11 +37,15 @@ final class SemanticTokenProvider(
   )
   cp.typeCheck(unit) // initializing unit
   val (root, source) = (unit.lastBody, unit.source)
+  implicit val ord: Ordering[NodeInfo] = Ordering.fromLessThan((ni1, ni2) =>
+    if (ni1.pos.start == ni2.pos.start) ni1.pos.end < ni2.pos.end
+    else ni1.pos.start < ni2.pos.start
+  )
 
   def unitPos(offset: Int): Position = unit.position(offset)
   val nodes: List[NodeInfo] = traverser
-    .traverse(List.empty[NodeInfo], root)
-    .sortBy(_.pos.start)
+    .traverse(SortedSet.empty[NodeInfo], root)
+    .toList
```

Comment:
why not Vector? :)

Reply:
I will try that :)

Reply:
I did some benchmarks on `Vectors` and the result are very similar to those using `List`. There is still a lot of place for improvements, but for now I would use these changes and then try to improve it more